### PR TITLE
Pagination on delete (followup)

### PR DESCRIPTION
**Pull request by keithjgrant**
##### SUMMARY
Introduces util function to help with pagination adjustment after deleting items in a list, to navigate back a page if all items on the current page have been deleted.

About half of the list components have been updated to use the new hook.

Addresses #6002 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/screens/Job/JobList/JobList.jsx [hunk 1]
```diff
@@ -34,257 +35,217 @@ const QS_CONFIG = getQSConfig(
...
+        {i18n._(t`Failed to delete one or more jobs.`)}
+        <ErrorDetail error={deletionError} />
+      </AlertModal>
+    </PageSection>
+  );
 }
 
-export { JobList as _JobList };
-export default withI18n()(withRouter(JobList));
+// export { JobList as _JobList };
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Job/JobList/JobList.jsx]: Delete this

File: awx/ui_next/src/screens/Job/JobList/JobList.test.jsx [hunk 2]
```diff
@@ -87,93 +87,161 @@ UnifiedJobsAPI.read.mockResolvedValue({
...
+}
 
-    done();
+describe('<JobList />', () => {
+  test('initially renders succesfully', async () => {
+    let wrapper;
+    await act(async () => {
+      wrapper = mountWithContexts(<JobList />);
+    });
+    await waitForLoaded(wrapper);
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Job/JobList/JobList.test.jsx:104]: You call this in each test.  Why not just do this inside of a `beforeEach || beforeAll`?  Your call.

File: awx/ui_next/src/util/useRequest.js [hunk 3]
```diff
@@ -1,4 +1,10 @@
 import { useEffect, useState, useRef, useCallback } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+import {
+  parseQueryString,
+  replaceParams,
+  encodeNonDefaultQueryString,
+} from './qs';
 
 /*
```
Comment (AlexSCorey) [inline awx/ui_next/src/util/useRequest.js]: Delete?

File: awx/ui_next/src/util/useRequest.js [hunk 4]
```diff
@@ -47,3 +53,40 @@ export default function useRequest(makeRequest, initialValue) {
...
+    }
+  }, [error]);
+
+  const deleteItems = async () => {
+    await request();
+    const params = parseQueryString(qsConfig, location.search);
+    if (params.page > 1 && allItemsSelected) {
+      const newParams = encodeNonDefaultQueryString(
+        qsConfig,
+        replaceParams(params, { page: params.page - 1 })
```
Comment (AlexSCorey) [inline awx/ui_next/src/util/useRequest.js:78]: nice work here.  I like the idea of just replacing the params with the proper params.  Simple

**Review by AlexSCorey — APPROVED** [review]

**Review by marshmalien — APPROVED** [review]
Very cool hook. I like how we were able to utilize useRequest within the new useDeleteItems hook.

**Pull request merged**